workflow-service/src/main/java/com/redhat/parodos/workflow/execution/repository/WorkFlowRepository.java

```diff
@@ -55,6 +55,9 @@ WorkFlowExecution findFirstByMainWorkFlowExecutionIdAndWorkFlowDefinitionId(UUID
 	@Query("SELECT w FROM prds_workflow_execution w WHERE w.mainWorkFlowExecution.id = :mainWorkflowId and EXISTS (SELECT f.type FROM prds_workflow_definition f WHERE f.id = w.workFlowDefinition.id AND f.type = com.redhat.parodos.workflow.enums.WorkFlowType.CHECKER)")
 	List<WorkFlowExecution> findCheckers(@Param("mainWorkflowId") UUID mainWorkflowId);
 
+	@Query("SELECT COUNT(*) FROM prds_workflow_execution o JOIN prds_workflow_execution restarted ON o.id = restarted.originalWorkFlowExecution.id WHERE o.id = :originalWorkflowId")
+	Integer countRestartedWorkflow(@Param("originalWorkflowId") UUID originalWorkflowId);
```

Comment:
pls add  a unit-test for it in `WorkFlowRepositoryTest`

Reply:
ah, good catch! :)